featurebyte/common/retry.py

```diff
@@ -0,0 +1,57 @@
+"""
+This module contains a decorator to retry a coroutine function
+"""
+from typing import Any, Callable, Optional, Tuple, Type, Union
+
+import asyncio
+import random
+from functools import wraps
+
+
+def async_retry(
+    max_retries: int = 3,
+    exceptions: Tuple[Type[BaseException], ...] = (Exception,),
+    backoff_factor: float = 2,
+    exception_signature_check: Optional[Callable[[BaseException], bool]] = None,
+) -> Callable[[Callable[..., Any]], Callable[..., Any]]:
+    """
+    Retry a coroutine function
+
+    Parameters
+    ----------
+    max_retries: int
+        Maximum number of retries
+    exceptions: Tuple[Type[BaseException], ...]
```

Comment:
might be worth adding a test to demonstrate how the `exceptions` and `exception_signature_check` params work